src/layers/core/icon-layer/icon-layer-vertex-64.glsl.js

```diff
@@ -36,6 +36,7 @@ attribute vec2 instanceOffsets;
 uniform vec2 viewportSize;
 uniform float sizeScale;
 uniform vec2 iconsTextureDim;
+uniform float devicePixelRatio;
```

Comment:
@rivulet-zhang , I don't see any code that sets up`devicePixelRatio` uniform, is this done inside any base class?
Also we recently added code so that apps provide `devicePixelRatio` as a prop to `DeckGL` component (#865), if this uniform setting depends on that code, we can't pick this PR for 4.1 patch release as #865 is not in current 4.1.1 patch. If yes, we have to remove dependency on #865.

Reply:
so this is setup as part of project module, should be good.